packages/synapse-react-client/src/utils/hooks/useDetectSSOCode.ts

```diff
@@ -87,105 +89,108 @@ export default function useDetectSSOCode(
     return null
   }, [isHandlingSynapseOAuthSignIn, searchParams])
 
-  const { accessToken } = useSynapseContext()
   const [isLoading, setIsLoading] = useState(!!(code && provider))
 
   useEffect(() => {
-    if (code && provider) {
-      const redirectUrl = `${redirectURL}?provider=${provider}`
+    if (hasInitializedSession) {
+      if (code && provider) {
+        const redirectUrl = `${redirectURL}?provider=${provider}`
 
-      //If user is already logged in, and the provider is ORCID, then try to bind this OAuth provider to the account.
-      if (OAUTH2_PROVIDERS.ORCID == provider && accessToken !== undefined) {
```

Comment:
And look at prop.token instead of SynapseContext.accessToken

Reply:
Is `token` passed as an arg ever different from `accessToken` retrieved through context? In `ApplicationSessionManager`, these are derived from the exact same variable